internal/services/cognitive/cognitive_account_rai_blocklist_resource.go

```diff
@@ -0,0 +1,216 @@
+		},
+
+		"cognitive_account_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: raiblocklists.ValidateAccountID,
+		},
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+	}
+}
+
+func (c CognitiveRaiBlocklistResource) Attributes() map[string]*schema.Schema {
+	return map[string]*schema.Schema{}
+}
+
+func (c CognitiveRaiBlocklistResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model cognitiveRaiBlocklistModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.Cognitive.RaiBlocklistsClient
+			accountId, err := raiblocklists.ParseAccountID(model.CognitiveAccountId)
+			if err != nil {
+				return err
+			}
+
+			locks.ByID(accountId.ID())
+			defer locks.UnlockByID(accountId.ID())
+
+			id := raiblocklists.NewRaiBlocklistID(accountId.SubscriptionId, accountId.ResourceGroupName, accountId.AccountName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(c.ResourceType(), id)
+			}
+
+			properties := &raiblocklists.RaiBlocklist{
+				Properties: &raiblocklists.RaiBlocklistProperties{},
+			}
+
+			if model.Description != "" {
+				properties.Properties.Description = &model.Description
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, *properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (c CognitiveRaiBlocklistResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model cognitiveRaiBlocklistModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.Cognitive.RaiBlocklistsClient
+			accountId, err := raiblocklists.ParseAccountID(model.CognitiveAccountId)
+			if err != nil {
+				return err
+			}
+
+			locks.ByID(accountId.ID())
+			defer locks.UnlockByID(accountId.ID())
+
+			id, err := raiblocklists.ParseRaiBlocklistID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return err
+			}
+
+			properties := existing.Model
+
+			if metadata.ResourceData.HasChange("description") {
+				properties.Properties.Description = pointer.To(model.Description)
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, *id, *properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
```

Comment:
We need to nil check Model and Properties here since we there needs to be something there for us to update, can you please update this to
```suggestion
			if existing.Model == nil {
				return fmt.Errorf("retrieving %s: `model` was nil", id)
			}
			if existing.Model.Properties == nil {
				return fmt.Errorf("retrieving %s: `properties` was nil", id)
			}
			
			payload := existing.Model

			if metadata.ResourceData.HasChange("description") {
				payload.Properties.Description = pointer.To(model.Description)
			}

			if _, err := client.CreateOrUpdate(ctx, *id, *payload); err != nil {
				return fmt.Errorf("updating %s: %+v", id, err)
			}
```